yt/utilities/on_demand_imports.py

```diff
@@ -249,6 +249,55 @@ def __getattr__(self, attr):
 _pooch = pooch_imports()
 
 
+class pyart_imports:
```

Comment:
Note: I worked out a way to simplify this part in https://github.com/yt-project/yt/pull/3935
If my PR gets merged first (or ever), I'm happy to help rewriting this (mostly) boilerplate code